core/src/main/java/hudson/model/UsageStatistics.java

```diff
@@ -126,51 +125,19 @@ private RSAPublicKey getKey() {
      * Gets the encrypted usage stat data to be sent to the Hudson server.
      */
     public String getStatData() throws IOException {
-        Jenkins j = Jenkins.getInstance();
-
         JSONObject o = new JSONObject();
-        o.put("stat",1);
-        o.put("install", j.getLegacyInstanceId());
-        o.put("servletContainer", j.servletContext.getServerInfo());
-        o.put("version", Jenkins.VERSION);
-
-        List<JSONObject> nodes = new ArrayList<JSONObject>();
-        for( Computer c : j.getComputers() ) {
-            JSONObject  n = new JSONObject();
-            if(c.getNode()==j) {
-                n.put("master",true);
-                n.put("jvm-vendor", System.getProperty("java.vm.vendor"));
-                n.put("jvm-name", System.getProperty("java.vm.name"));
-                n.put("jvm-version", System.getProperty("java.version"));
-            }
-            n.put("executors",c.getNumExecutors());
-            DescriptorImpl descriptor = j.getDescriptorByType(DescriptorImpl.class);
-            n.put("os", descriptor.get(c));
-            nodes.add(n);
-        }
-        o.put("nodes",nodes);
-
-        List<JSONObject> plugins = new ArrayList<JSONObject>();
-        for( PluginWrapper pw : j.getPluginManager().getPlugins() ) {
-            if(!pw.isActive())  continue;   // treat disabled plugins as if they are uninstalled
-            JSONObject p = new JSONObject();
-            p.put("name",pw.getShortName());
-            p.put("version",pw.getVersion());
-            plugins.add(p);
-        }
-        o.put("plugins",plugins);
-
-        JSONObject jobs = new JSONObject();
-        List<TopLevelItem> items = j.getAllItems(TopLevelItem.class);
-        for (TopLevelItemDescriptor d : Items.all()) {
-            int cnt=0;
-            for (TopLevelItem item : items) {
-                if(item.getDescriptor()==d)
-                    cnt++;
+        for (UsageStatisticsContributor contributor : UsageStatisticsContributor.all()) {
+            JSONObject contributed = contributor.getStatData();
```

Comment:
@daniel-beck OK, on it now. 

Question about robustness: do you think the code should just log an error if a `UsageStatisticsContributor` tries to override a key instead of throwing an IllegalStateException as it does now? 

IMO, to be consistent with catching exception here, it would seem more logical to try and continue instead in general?